runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -10342,31 +10367,35 @@ static TR::Register* inlineIntrinsicStringIndexOfString(TR::Node* node, TR::Code
    generateRegRegInstruction(TR::InstOpCode::CMP4RegReg, node, resultReg, maxReg, cg);
    generateLabelInstruction(TR::InstOpCode::JG4, node, notFoundLabel, cg);
 
-   generateRegMemInstruction(TR::InstOpCode::LEARegMem(), node, tmpReg, generateX86MemoryReference(s1Reg, resultReg, 0, hdrSize, cg), cg);
+   generateRegMemInstruction(TR::InstOpCode::LEARegMem(), node, tmpReg, generateX86MemoryReference(s1Reg, resultReg, shift, hdrSize, cg), cg);
    generateRegRegInstruction(TR::InstOpCode::MOV4RegReg, node, ECX, tmpReg, cg);
    generateRegImmInstruction(TR::InstOpCode::AND4RegImms, node, ECX, width - 1, cg);
    generateLabelInstruction(TR::InstOpCode::JE1, node, firstCharLoopLabel, cg);
 
    generateRegImmInstruction(TR::InstOpCode::ANDRegImms(), node, tmpReg, ~(width - 1), cg);
-   generateRegMemInstruction(TR::InstOpCode::MOVDQURegMem, node, xmmReg1, generateX86MemoryReference(tmpReg, 0, cg), cg);
-   generateRegRegInstruction(TR::InstOpCode::PCMPEQBRegReg, node, xmmReg1, xmmReg2, cg);
+   generateRegMemInstruction(TR::InstOpCode::MOVUPSRegMem, node, xmmReg1, generateX86MemoryReference(tmpReg, 0, cg), cg);
```

Comment:
Why did you change MOVDQU to MOVUPS here and elsewhere?  Sorry, the reason isn't apparent to me.

Reply:
I was comparing the performance with MOVDQU and with MOVUPS, and found no particular difference between the two.
Shall I revert it to MOVUPS?

Reply:
I don't think there will be an observable difference between these two instructions in this context. But I think it is more conventional to use `MOVDQU` (as it was originally) when dealing with byte/short/int data. `MOVUPS` is usually used for floating point data.

Reply:
Changed back to MOVDQU.